include/swift/ClangImporter/CXXMethodBridging.h

```diff
@@ -8,70 +8,92 @@
 #include <string>
 namespace swift {
 struct CXXMethodBridging {
-  enum class Kind { unkown, getter, setter, subscript };
+  enum class Kind { unknown, getter, setter, subscript };
 
-  enum class NameKind { unkown, snake, lower, camel, title };
+  enum class NameKind { unknown, snake, lower, camel, title };
 
   CXXMethodBridging(const clang::CXXMethodDecl *method) : method(method) {}
 
   Kind classify() {
     if (nameIsBlacklist())
-      return Kind::unkown;
+      return Kind::unknown;
 
     // this should be handled as snake case. See: rdar://89453010
     // case. In the future we could
     //  import these too, though.
     auto nameKind = classifyNameKind();
     if (nameKind != NameKind::title && nameKind != NameKind::camel &&
-        nameKind != NameKind::lower)
-      return Kind::unkown;
+        nameKind != NameKind::lower && nameKind != NameKind::snake)
+      return Kind::unknown;
 
     if (getClangName().startswith_insensitive("set")) {
       // Setters only have one parameter.
       if (method->getNumParams() != 1)
-        return Kind::unkown;
+        return Kind::unknown;
 
       // rdar://89453106 (We need to handle imported properties that return a
       // reference)
       if (method->getParamDecl(0)->getType()->isReferenceType())
-        return Kind::unkown;
+        return Kind::unknown;
 
       return Kind::setter;
     }
 
     // Getters and subscripts cannot return void.
     if (method->getReturnType()->isVoidType())
-      return Kind::unkown;
+      return Kind::unknown;
 
     if (getClangName().startswith_insensitive("get")) {
       // Getters cannot take arguments.
       if (method->getNumParams() != 0)
-        return Kind::unkown;
+        return Kind::unknown;
 
       // rdar://89453106 (We need to handle imported properties that return a
       // reference)
       if (method->getReturnType()->isReferenceType())
-        return Kind::unkown;
+        return Kind::unknown;
 
       return Kind::getter;
     }
 
     // rdar://89453187 (Add subscripts clarification to CXXMethod Bridging to
     // clean up importDecl)
-    return Kind::unkown;
+    return Kind::unknown;
   }
 
   NameKind classifyNameKind() {
     bool allLower = llvm::all_of(getClangName(), islower);
+    bool hasUpper = false;
 
-    if (getClangName().empty())
-      return NameKind::unkown;
+    for (std::size_t i = 0; i < getClangName().size(); i++) {
+      if (std::isupper(getClangName()[i])) {
+        hasUpper = true;
+      }
+    }
 
-    if (getClangName().contains('_'))
-      return allLower ? NameKind::snake : NameKind::unkown;
+    // if the string is empty
+    if (getClangName().empty()) {
+      return NameKind::unknown;
+    }
 
-    if (allLower)
-      return NameKind::lower;
+    // if one of the elements is an underscore
+    if (getClangName().contains('_')) {
+      if (allLower) {
```

Comment:
I think we can get rid of `allLower` all together. Just do `return hasUpper ? unknown : snake`.